### PR TITLE
Searching Algorithm: Ternary Search

### PR DESCRIPTION
**Pull request by atarax665**
Added ternary search algorithm with proper in-code comments and a beginner introduction of the algorithm. I am planning to go for a while with this repository and implement few more algos in Ruby. Will be looking forward for some suggestions to improve my code.

### CODE REVIEW

**Review by vbrazo — COMMENTED** [review]
Thanks for the PR :)

We need a new entry in the DIRECTORY file in order to have proper documentation for this new algorithm

**Comment by vbrazo** [comment]
I'm also seeing that we don't have an explanation for this algorithm in this repo https://github.com/TheAlgorithms/Algorithms-Explanation. It'd be nice if you could add a more detailed description of how the algorithm works.

File: searches/ternary_search.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Ternary search is a searching technique that is used to search the position of a specific value in an array.
+Ternary search is a divide-and-conquer algorithm.
+It is mandatory for the array to be sorted (in which you will search for an element).
+The array is divided into three parts and then we determine in which part the element exists.
+In this search, after each iteration it neglects 1/3 part of the array and repeats the same operations on the remaining ⅔.
+Time Complexity: O(log3 n)
+Space Complexity: O(1)
+=end
+
+def ternary_search(l, r, key, arr)
```
Comment (vbrazo) [inline searches/ternary_search.rb:13]:
```suggestion
def ternary_search(left, right, key, arr)
```

File: searches/ternary_search.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    end
+end
+
+puts "Enter a space-separated list:"
+arr = gets.chomp.split(' ').map(&:to_i)
+puts "Enter a value to be searched:"
+key = gets.chomp.to_i
+puts ternary_search(0, arr.length - 1, key, arr) != nil ?
+  "Found at index #{ternary_search(0, arr.length - 1, key, arr)}" :
+  "Not found"
```
Comment (vbrazo) [inline searches/ternary_search.rb:43]: This kind of test works but it'd be great if we could have a Minitest script.

Comment (atarax665) [reply searches/ternary_search.rb:43]: Should I replace the dynamic input test with a minitest script or I should add the minitest at the end of the program in commented form?

**Comment by atarax665** [comment]
> Thanks for the PR :)
> 
> We need a new entry in the DIRECTORY file in order to have proper documentation for this new algorithm

Its read in the contribution guidelines that the DIRECTORY.md is not to be updated manually by the contributors. (```Do not update the README.md or DIRECTORY.md file which will be periodically autogenerated by our Travis CI processes.```) Should I update it manually?

**Comment by vbrazo** [comment]
Yeah you are right. Please disregard my comment.

**Comment by atarax665** [comment]
> I'm also seeing that we don't have an explanation for this algorithm in this repo https://github.com/TheAlgorithms/Algorithms-Explanation. It'd be nice if you could add a more detailed description of how the algorithm works there.

Sure, I will be raising a PR regarding this soon.

**Review by vbrazo — APPROVED** [review]

File: searches/ternary_search.rb (hunk 2 above)
Comment (vbrazo) [reply searches/ternary_search.rb:43]: a Minitest script would be cool because the CI runs the tests

**Pull request merged**